src/promptflow/promptflow/_cli/_pf_azure/entry.py

```diff
@@ -113,13 +117,14 @@ def entry(argv):
         setup_user_agent_to_operation_context(args.user_agent)
     logger = get_telemetry_logger()
     custom_dimensions = _get_workspace_info(args)
-    with log_activity(
-        logger,
-        _get_cli_activity_name(cli=prog, args=args),
-        activity_type=ActivityType.PUBLICAPI,
-        custom_dimensions=custom_dimensions,
-    ):
-        run_command(args)
+    with update_user_agent_context_according_to_flow_language(args):
```

Comment:
can we implement this without reading flow/run in the begining?
Maybe modify some context inside?